File: git-recon.go
// recon.go
// NOTE: This Go script is for domain-based recon.
// GitHub secrets scanning tools like TruffleHog, Gitleaks, etc., are added in a separate Bash script below.

package main

// [Go recon code remains unchanged here...]

/*
===========================
 GitHub Secrets Scanner Script
===========================

#!/bin/bash

if [ "$#" -ne 1 ]; then
  echo -e "\033[93mUsage: $0 <github-repo-or-org>\033[0m"
  echo -e "\033[93mExample: $0 https://github.com/org/repo\033[0m"
  exit 1
fi

REPO=$1

GREEN="\033[92m"
YELLOW="\033[93m"
RED="\033[91m"
RESET="\033[0m"

mkdir -p github_secrets_output
cd github_secrets_output

# Clone repo
echo -e "${YELLOW}[+] Cloning $REPO${RESET}"
git clone --depth=1 $REPO repo 2>/dev/null
cd repo || exit

# TruffleHog
echo -e "${GREEN}[+] Running TruffleHog...${RESET}"
trufflehog filesystem . > ../trufflehog.txt

# Gitleaks
echo -e "${GREEN}[+] Running Gitleaks...${RESET}"
gitleaks detect --source=. --report-format=json --report-path=../gitleaks.json

# git-all-secrets
echo -e "${GREEN}[+] Running git-all-secrets...${RESET}"
git-all-secrets -r . -o ../gitallsecrets.txt --no-banner

# detect-secrets
echo -e "${GREEN}[+] Running detect-secrets...${RESET}"
detect-secrets scan > ../detect-secrets.json

cd ..

# Color-coded summary
for f in trufflehog.txt gitleaks.json gitallsecrets.txt detect-secrets.json; do
  echo -e "\n${YELLOW}[*] Showing results from: $f${RESET}"
  cat "$f" | head -n 20
  echo -e "${RED}...Output truncated... See full in github_secrets_output/$f${RESET}"
done

===========================
 Installation Instructions
===========================

# TruffleHog
pip install trufflehog

# Gitleaks
brew install gitleaks  # or
curl -s https://raw.githubusercontent.com/gitleaks/gitleaks/main/install.sh | bash

# git-all-secrets
pip install git+https://github.com/anshumanbh/git-all-secrets.git

# detect-secrets
pip install detect-secrets
*/

File: gitrecon.py
# recon.py
# NOTE: This Python script is for GitHub secrets scanning using popular open-source tools.

import os
import subprocess
import sys
import shutil

GREEN = "\033[92m"
YELLOW = "\033[93m"
RED = "\033[91m"
RESET = "\033[0m"

TOOLS = [
    ("trufflehog", ["trufflehog", "filesystem", "."], "trufflehog.txt"),
    ("gitleaks", ["gitleaks", "detect", "--source=.", "--report-format=json", "--report-path=../gitleaks.json"], "gitleaks.json"),
    ("git-all-secrets", ["git-all-secrets", "-r", ".", "-o", "../gitallsecrets.txt", "--no-banner"], "gitallsecrets.txt"),
    ("detect-secrets", ["detect-secrets", "scan"], "detect-secrets.json"),
]

def clone_repo(repo_url):
    print(f"{YELLOW}[+] Cloning {repo_url}{RESET}")
    os.makedirs("github_secrets_output", exist_ok=True)
    os.chdir("github_secrets_output")
    if os.path.exists("repo"):
        shutil.rmtree("repo")
    subprocess.run(["git", "clone", "--depth=1", repo_url, "repo"], stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
    if not os.path.exists("repo"):
        print(f"{RED}Failed to clone repository.{RESET}")
        sys.exit(1)
    os.chdir("repo")

def run_tools():
    for name, cmd, outfile in TOOLS:
        print(f"{GREEN}[+] Running {name}...{RESET}")
        with open(f"../{outfile}", "w") as f:
            try:
                subprocess.run(cmd, stdout=f, stderr=subprocess.DEVNULL, text=True)
            except Exception as e:
                print(f"{RED}Error running {name}: {e}{RESET}")

def summarize_results():
    os.chdir("..")
    for filename in [x[2] for x in TOOLS]:
        print(f"\n{YELLOW}[*] Showing results from: {filename}{RESET}")
        try:
            with open(filename, "r") as f:
                lines = f.readlines()[:20]
                for line in lines:
                    print(line.strip())
                print(f"{RED}...Output truncated... See full in github_secrets_output/{filename}{RESET}")
        except FileNotFoundError:
            print(f"{RED}File not found: {filename}{RESET}")

def main():
    if len(sys.argv) != 2:
        print(f"{YELLOW}Usage: python3 recon.py <github-repo-url>{RESET}")
        print(f"{YELLOW}Example: python3 recon.py https://github.com/org/repo{RESET}")
        sys.exit(1)

    repo_url = sys.argv[1]
    clone_repo(repo_url)
    run_tools()
    summarize_results()

if __name__ == "__main__":
    main()

"""
Installation Instructions:

pip install trufflehog detect-secrets
brew install gitleaks   # or use curl install script
pip install git+https://github.com/anshumanbh/git-all-secrets.git
"""
